### PR TITLE
Showing collection code after POST meal

### PR DESCRIPTION
**Pull request by Gmiller290488**
I added code to present an alert displaying a collection code if a Meal is successfully added.
Otherwise show an error message.

### CODE REVIEW

File: iosHackthonApp/iosHackthonApp/Add Meal Screen/AddMealView.swift [hunk 1]
```diff
@@ -11,45 +11,45 @@ import shared
...
-                quantity: 2,
-                availableFromDate: "Tuesday",
-                expiryDate: "Saturday",
-                info: "Meat",
-                hot: false,
-                locationLat: 51.509865,
-                locationLong:  -0.118092)
-            self.viewModel.postMeal(meal: meal)
+        ZStack {}
+        .alert(isPresented: $viewModel.showingCollectionCode) {
```
Comment (Gmiller290488) [inline iosHackthonApp/iosHackthonApp/Add Meal Screen/AddMealView.swift:18]: SwiftUI gets upset if you have two alerts on the same view so this is just here temporarily.
When we get to the stage of showing something more elaborate, we can then get rid of this ZStack and alert code.

File: iosHackthonApp/iosHackthonAppTests/Strings+extensionTests.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    func test_last4Chars_asNumbers_returnsCorrectString() {
+        XCTAssertEqual("abcg1212".last4Chars(), "1212")
+    }
+
+    func test_last4Chars_asCombined_returnsCorrectString() {
+        XCTAssertEqual("abcg12".last4Chars(), "CG12")
+    }
+}
```
Comment (dsmith42) [inline iosHackthonApp/iosHackthonAppTests/Strings+extensionTests.swift:25]: Nice - we have some tests 😀

**Review by dsmith42 — APPROVED** [review]
Looks good to me👍

File: iosHackthonApp/iosHackthonAppTests/Strings+extensionTests.swift (hunk 2 above)
Comment (Gmiller290488) [reply iosHackthonApp/iosHackthonAppTests/Strings+extensionTests.swift:25]: Only a few 🙈

**Pull request merged**